chainer/functions/connection/multihead_attention.py

```diff
@@ -0,0 +1,286 @@
+    proj_out_b=None,  # type: variable.Variable
+    add_zero_attn=False,  # type: bool
+    dropout=0,  # type: float
+    post_dropout=0,  # type: float
+    key_padding_mask=None,  # type: tp.Optional[InputType]
+    attn_mask=None,  # type: tp.Optional[InputType]
+    scaling=None,  # type: tp.Optional[float]
+    return_weights=True  # type: tp.Optional[bool]
+):
+    # type: (...) -> tp.Union[variable.Variable, tp.Tuple[variable.Variable, variable.Variable]]  # NOQA
+    """Multi-head Attention forward function.
+
+    Args:
+        query (:class:`~chainer.Variable` or :ref:`ndarray`):
+            A batch of query vectors whose shape is :math:`(L, B, E)` where
+            :math:`L` is the target sequence length, :math:`B` is the
+            batch size, and :math:`E` is the embedding size.
+        key (:class:`~chainer.Variable` or :ref:`ndarray`)
+            A batch of key vectors whose shape is :math:`(S, B, E)` where
+            :math:`S` is the source sequence length, :math:`B` is the
+            batch size, and :math:`E` is the embedding size.
+        value (:class:`~chainer.Variable` or :ref:`ndarray`)
+            A batch of value vectors whose shape is :math:`(S, B, E)` where
+            :math:`S` is the source sequence length, :math:`B` is the
+            batch size, and :math:`E` is the embedding size.
+        expected_embedding_size (int): Total number of units of the model.
+        n_head (int): The number of parallel attention heads.
+        proj_in_W (tuple of :class:`~chainer.Variable`\\s, :class:`~chainer.Variable` or :ref:`ndarray`):
+            Weight(s) to project `query`, `key`, and `value` vectors.
+            If the input sizes of `query`, `key`, and `value` are different,
+            this should be the tuple of 3 weights, otherwise, 1 weight.
+        proj_in_b (:class:`~chainer.Variable` or :ref:`ndarray`):
+            Bias added to projected `query`, `key`, and `value` vectors.
+        add_zero_attn (bool): If ``True``, add a new batch of zeros to
+            the key and value sequences at axis=1.
+        dropout (float): Dropout ratio at the attention layer.
+        post_dropout (float): Dropout ratio at the output.
+        proj_out_W (:class:`~chainer.Variable` or :ref:`ndarray`)
+        proj_out_b (:class:`~chainer.Variable` or :ref:`ndarray`)
+        key_padding_mask (:class:`~chainer.Variable` or :ref:`ndarray`):
+            If not ``None``, specified padding elements in the key
+            will be ignored by the attention.
+            The shape is :math:`(B, S)` where :math:`B` is the batch size,
+            and :math:`S` is the source sequence length.
+        attn_mask (:class:`~chainer.Variable` or :ref:`ndarray`):
+            Mask help attention ignores certain positions.
+            The shape is :math:`(L, L)` where :math:`L` is
+            the target sequence length.
+        return_weights (bool): If ``True``, return averaged attention weights.
+
+    Outputs:
```

Comment:
Fix this by mimicking https://github.com/chainer/chainer/blob/master/chainer/functions/connection/n_step_rnn.py#L443

> 
    Returns:
        tuple: This function returns a tuple containing three elements,
        ``hy`` and ``ys``.
        - ``hy`` is an updated hidden states whose shape is same as ``hx``.
        - ``ys`` is a list of :class:`~chainer.Variable` . Each element
          ``ys[t]`` holds hidden states of the last layer corresponding
          to an input ``xs[t]``. Its shape is ``(B_t, N)`` where ``B_t`` is
          mini-batch size for time ``t``, and ``N`` is size of hidden
          units. Note that ``B_t`` is the same value as ``xs[t]``.